### ISSUE TITLE
Second model

### ISSUE THREAD

**Issue by GreNait** [GreNait/technical-thesis#13]
- [x] add gaussian noise and find out the performance in comparison to a model without the gaussian noise
- [x] train first without any optimization
- [x] use validation data for verifying the model
- [x] show training accuracy in plots for including this into the status quo presentation
- [x] find out what the prediction of -30 means within the model
- [x] worth investigating - https://github.com/keras-team/keras/issues/3477 - https://stackoverflow.com/questions/46377940/proper-way-to-make-prediction-with-keras-model-trained-with-imagedatagenerator

**Comment by GreNait** [comment]
I am starting the second model based on the sample on the tensorflow homepage. https://www.tensorflow.org/tutorials/images/classification

![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e5e39ab4cd4ed0011ab0e15)

![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e5e39bab0320f0012e54fed)

**Comment by GreNait** [comment]
Training the second model lead the first time directly to overfitting

![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e5e3c4c798f1d0015410f8c)

![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e5e3f1e8f64390011ee9dcc)

**Comment by GreNait** [comment]
Splitting the data helps a lot. Using the ImageDataGenerator for splitting the data into train and validation imporved the model accuracy. The Dropout might also helkp against the overfitting. The second model reached an accuracy for 99,4%

![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e5e708d8f64390011eea981)

![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e5e70a5b0320f0012e55c33)

**Comment by GreNait** [comment]
The perfomence difference between the using noise in iomages wasn't that much. Well, at least not in the validation images. A usefull test set up has to be created.

![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e5f8d98b0320f0012e59125)

**Comment by GreNait** [comment]
I tried a different model with different layers. The training was longer (some minutes) but the overfitting is dramatic. I am not sure, if this is because of the model layotu or of a different reason.

![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e6533b0ae732d00112921e2)

![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e6533c5916ce50012849c2f)
![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e6533e4b3308f0011cc9a39)

**Comment by GreNait** [comment]
I changed to a SparseCategoricalCrossentropy (Based on the sample on tensorflow for obect classification with clothes) in hope to get a better prediction of batman and deadpool.

The accuracy was quite high, again, an indication for overfitting.![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e653a95916ce50012849d0e)

**Comment by GreNait** [comment]
It might be, that relu is not good for my images. See https://datascience.stackexchange.com/questions/5706/what-is-the-dying-relu-problem-in-neural-networks

I am trying now the sigmoid function. Perhaps that works better.

**Comment by GreNait** [comment]
Using sigmoid instead of relu had a dramatic effect on the training and validation graphs. I am not sure, what this means. Also the accuracy is arround 50%. Whatever that means...

![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e656770916ce5001284a29c)

**Comment by GreNait** [comment]
Using a image data generator also for testing the data with model.predict_generator worked, but the values are the same for all images. Appreentyl
 ![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e6df681ec64f000118f268d)

**Comment by GreNait** [comment]
A better trained (small) model together with the tf.keras.Sequential([model, tf.keras.layers.Softmax()]) works for the predictions. Even with the smartphone image.

This is the test data directory structure
![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e6e03641332eb001111a19e)

This is the result using the prediction:
![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e6e039931df0b001242592d)

I am hoping, that the 0 is a batman and the 1 a deadpool. Next test is the data in one folder.

**Comment by GreNait** [comment]
In one folder also works

![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e6e04c7ec64f000118f27ca)

![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e6e04d81332eb001111a1ba)
